app/models/course_data/courses_users.rb

```diff
@@ -118,6 +119,13 @@ def character_sum(revisions, namespace)
       .sum(:characters) || 0
   end
 
+  def references_sum(revisions)
+    revisions
+      .where(articles: { namespace: Article::Namespaces::MAINSPACE, deleted: false })
+      .to_a
+      .sum(&:references_added) || 0
```

Comment:
I think both the `.to_a` and the `|| 0` are not necessary.

Even though `revisions` is an ActiveRecord relation (rather than an array), the `.sum` method will fall back to the array method if it receives a block argument rather than a column name: https://github.com/rails/rails/blob/b9ca94caea2ca6a6cc09abaffaad67b447134079/activerecord/lib/active_record/relation/calculations.rb#L86

`Array#sum` defaults to 0 for an empty array.